internal/dao/container.go

```diff
@@ -47,11 +46,15 @@ func (c *Container) List(ctx context.Context, _ string) ([]runtime.Object, error
 		return nil, err
 	}
 	res := make([]runtime.Object, 0, len(po.Spec.InitContainers)+len(po.Spec.Containers))
-	for _, co := range po.Spec.InitContainers {
-		res = append(res, makeContainerRes(co, po, cmx[co.Name], true))
+	for i, co := range po.Spec.InitContainers {
```

Comment:
I think the knowledge of the container type is right here i.e what kind of container are we dealing with? 
aka init, main, ephemeral (let's not forget those!). 
I think it's best to precompute all the necessary bits vs lazy eval when one needs the index or other info out of containerRes.
So I think its best (and less intrusive!) to let containerRes  be dumb but add the differentiation right here at the call site and thus mod the current signature to something like this:
```go
func makeContainerRes(idx string, co v1.Container, po *v1.Pod, cmx *mv1beta1.ContainerMetrics) render.ContainerRes {...}
```

Then at the call site:

```go
res = append(res, makeContainerRes("I"+strconv.Itoa(i+1), co, po, cmx[co.Name]))
```

I also think we should go with non-zero index and perhaps use the scheme I, M, E for init, main and ephemeral.